utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/netclass.lua

```diff
@@ -0,0 +1,77 @@
+local ubus = require "ubus"
+
+-- reference/further reading:
+-- - node_exporter netclass_linux (upstream metrics): https://github.com/prometheus/node_exporter/blob/master/collector/netclass_linux.go
+-- - relevant sysfs files: https://github.com/prometheus/procfs/blob/5f46783c017ef6a934fc8cfa6d1a2206db21401b/sysfs/net_class.go#L121
+-- - get devices / read files: https://github.com/openwrt/packages/blob/openwrt-21.02/utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/snmp6.lua
+
+local function get_devices() -- based on hostapd_stations.lua
+   local u = ubus.connect()
+  local status = u:call("network.device", "status", {})
+  local devices = {}
+
+  for dev, dev_table in pairs(status) do
+    table.insert(devices, dev)
+  end
+  return devices
+end
+
+local function load(device, file) -- load a single sysfs file, trim trailing newline, return "" on error
+  local success, data = pcall(function () return string.gsub(get_contents("/sys/class/net/" .. device .. "/" .. file), "\n$", "") end)
+  if success then
+    return data
+  else
+    return ""
```

Comment:
I think returning nil would be cleaner.

```suggestion
    return nil
```